NuRadioReco/modules/io/NuRadioRecoio.py

```diff
@@ -49,7 +50,7 @@ def __init__(self, filenames, parse_header=True, parse_detector=True, fail_on_ve
             filenames = [filenames]
 
         self.__file_scanned = False
-        self.logger = logging.getLogger('NuRadioReco.NuRadioRecoio')
+        self.logger = setup_logger('NuRadioReco.NuRadioRecoio')
```

Comment:
Indeed, individual modules should use the `getLogger` function from the standard `logging` library. Using the `setup_logger()` might work, but I have not tested what the result is.

Reply:
why wouldn't we use the nice new standardized logging? I have experienced no error so far. And I think this is exactly want we want to do to have a uniform formatting of logger outputs and the status level. Or why would you disagree @MijnheerD @sjoerd-bouma ?

Reply:
sry if I'm slow in understanding how it works. Doesn't it require that the script that imported NuRadioRecoIO initialized a logger with the setup_logger function before the NuRadioRecoIO module initializes its logger? 
And I thought if we use the correct naming scheme, i.e., starting the logger name with "NuRadioReco." it will inherit from the correct logger?

Reply:
Indeed, the inheritance is taken care of by the naming scheme. But the way the logging is currently set up, is such that the parent logger (i.e. the one with the name "NuRadioReco") is the one who actually handles the log messages. All module loggers simply pass on their messages to the parent logger. And it is the latter that the `setup_logger()` function is supposed to set up.

As the inheritance is something that comes from the standard `logging` module itself, there is no need to import anything from our own logging library to initialize the module loggers, they can be called using `logging.getLogger()` . When using the modules in a script, their loggers will link to the parent logger that is expected to be created at the beginning [^1] of the script (using the `setup_logger()` function).

[^1]: From my testing the ordering is actually not important, i.e. the NuRadioReco logger can be created after modules are initialized (though this is not recommended). The modules loggers will find the parent logger as long as it is exists.